### PR TITLE
Finished code challenge 12

### PR DESCRIPTION
**Pull request by PotatoGod123**
- [x ] Top-level README “Table of Contents” is updated
 - [ x] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
     - [ x] “Happy Path” - Expected outcome
     - [ x] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [x ] README for this challenge is complete
     - [x ] Summary, Description, Approach & Efficiency, Solution
     - [x ] Link to code
     - [ x] Picture of whiteboard

### CODE REVIEW

File: challenges/test/lib/src/main/java/code/utilities/AnimalShelter.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }else if(animal instanceof Cat){
+      ((Cat) animal).count=counter;
+      counter++;
+      cats.enqueue(animal);
+      System.out.println(((Cat) animal).name+" the Cat has been added to the shelter");
+    }
+  }
+
+  public T dequeue(String pref){
+    String lowered= pref.toLowerCase(Locale.ROOT);
```
Comment (Mattpet26) [inline challenges/test/lib/src/main/java/code/utilities/AnimalShelter.java:27]: Nice work, accounting for lower case is important!

File: challenges/test/lib/src/test/java/code/utilitiestest/AnimalShelterTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Cat catTest2 = new Cat("CattyTwo");
+    Dog dogTest3 = new Dog("PupperThree");
+    Cat catTest3 = new Cat("CattyThree");
+    testSheleter.enqueue(dogTest);
+    testSheleter.enqueue(catTest);
+    testSheleter.enqueue(dogTest2);
+    testSheleter.enqueue(catTest2);
+    testSheleter.enqueue(dogTest3);
+    testSheleter.enqueue(catTest3);
+    System.out.println(testSheleter);
```
Comment (Mattpet26) [inline challenges/test/lib/src/test/java/code/utilitiestest/AnimalShelterTest.java:25]: I'm noticing a lot of people using sout's instead of adding additional tests??
